server/src/Ogmios/Query.hs

```diff
@@ -0,0 +1,91 @@
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE RecordWildCards #-}
+
+module Ogmios.Query (
+  makeRequest,
+  tryQueryUntilZero,
+  RequestOption (Query, Acquire),
+  defaultServerParameters,
+  ServerParameters (..),
+  QueryOption (..),
+) where
+
+import Control.Exception.Base (SomeException, try)
+
+import Data.Bifunctor (first)
+import Data.ByteString.Lazy (ByteString)
+import Data.Text (Text)
+import Network.Socket (withSocketsDo)
+import Network.WebSockets (ClientApp)
+import Network.WebSockets qualified as WebSockets
+import System.IO (hFlush, stdout)
+import System.Time.Extra qualified as Time.Extra
+
+--------------------------------------------------------------------------------
+
+data QueryOption = CurrentProtocolParameters
+
+data RequestOption
+  = Query QueryOption
+  | -- | TODO : add point parameter to Acquire
+    Acquire
+
+data ServerParameters = ServerParameters
+  { getHost :: String
+  , -- | WebSockets needs the port as Int.
+    getPort :: Int
+  , getPath :: String
+  }
+
+defaultServerParameters :: ServerParameters
+defaultServerParameters =
+  ServerParameters
+    { getHost = "ogmios"
+    , getPort = 1337
+    , getPath = "/"
+    }
+
+getQueryName :: QueryOption -> Text
+getQueryName CurrentProtocolParameters = "currentProtocolParameters"
+
+buildRequestJSON :: RequestOption -> Text
+buildRequestJSON (Query CurrentProtocolParameters) =
+  "{ \"type\": \"jsonwsp/request\", \"version\": \"1.0\", \"servicename\": \"ogmios\","
+    <> "\"methodname\": \"Query\", \"args\": { \"query\":\""
+    <> getQueryName CurrentProtocolParameters
+    <> "\"} }"
+buildRequestJSON _ = error "Unsuported Ogmios request"
```

Comment:
Following my suggestion for removing `QueryOption` would make this a total function